feat(mojaloop/#3533): mojaloop helm release v15.2.0 (#581)
mojaloop/helm: .changelog/release-v15.2.0.md

# Helm Release Notes

Date | Revision | Description
---------|----------|---------
 2023-10-23 | 0 | Initial draft
 2023-10-31 | 1 | RC.1

- For *BREAKING CHANGES*, please review the section `#5` "Breaking Changes" below.
- For *KNOWN ISSUES*, please review the section `#8` "Known Issues" below.

## 0. Summary

Enhancements and non-breaking changes to the [v15.1.0 Release](https://github.com/mojaloop/helm/blob/master/.changelog/release-v15.1.0.md), which includes:

1. Nodejs version of core and supporting services upgraded to v18.17.1 LTS
2. Performance improvements and metrics around core services: ALS, Central Ledger, Quoting Service based on characterisation and fixes, enhancements done
3. General maintenance and bug fixes

## 1. New Features

> 1. **Performance fixes, enhancements and maintenance:**  

Reviewer: I just noticed that some of these updates were missed/swallowed: 

- mojaloop/#3519: add topic name override config map logic (https://github.com/mojaloop/central-ledger/issues/969) ([b99d091](https://github.com/mojaloop/central-ledger/commit/b99d09129f328aab60e4059035be4d263ecd9f98)), closes https://github.com/mojaloop/project/issues/3519 - https://github.com/mojaloop/central-ledger/releases/tag/v17.2.0
- mojaloop/#3498: alter message keys for prepare, fulfil and timeout (https://github.com/mojaloop/central-ledger/issues/965) ([db3e3b2](https://github.com/mojaloop/central-ledger/commit/db3e3b2f0450f1093f1f20eb4f49b4f4de53ddd0)), closes https://github.com/mojaloop/project/issues/3498 - from https://github.com/mojaloop/central-ledger/releases/tag/v17.1.0
- json stringify on every response (https://github.com/mojaloop/central-ledger/issues/961) ([6798bfd](https://github.com/mojaloop/central-ledger/commit/6798bfda51be18c2d44c22e10ef67bc76c8b2195)) - https://github.com/mojaloop/central-ledger/releases/tag/v17.0.5
- mojaloop/#2880: simplify liquidity cover check and fix issue with validation (for main) (https://github.com/mojaloop/central-ledger/issues/918) ([825faf7](https://github.com/mojaloop/central-ledger/commit/825faf7c1c918b12eddedeb97eea0a2d563457e4)) - https://github.com/mojaloop/central-ledger/releases/tag/v17.0.4

Its possible that we have missed others. I recommend double checking this list against the issue report for the PI to double double check.

Author: @aaronreynoza is looking into this and will update.

    - the Quoting service operating mode to be set quoting-service.config.simple_routing_mode_enabled=true (in helm mojaloop/values.yaml under quoting-service config). If this is incorrectly configured, it will result in several failures in the quoting-service tests (7 expected failures). If this is disabled, ensure that you update the corresponding test-case environment variable parameter **SIMPLE_ROUTING_MODE_ENABLED** ( in helm mojaloop/values.yaml ml-testing-toolkit -> extraEnvironments.hub-k8s-default-environment.json.inputValues) to match.
    - the **on-us transfers** (in mojaloop/values.yaml "enable_on_us_transfers: false" under centralledger-handler-transfer-prepare -> config and  cl-handler-bulk-transfer-prepare -> config) configuration to be disabled. The test-case environment variable parameter (**ON_US_TRANSFERS_ENABLED** (in helm mojaloop/values.yaml ml-testing-toolkit -> extraEnvironments.hub-k8s-default-environment.json.inputValues), the same name used on postman collections) must similarly match this value.

> 4. Simulators

Reviewer 2: ```suggestion
## 4. Simulators
```

8. Testing the new Bulk functionality (sdk-scheme-adapter)

    For details regarding deployment and validation of simulators needed for bulk (for adoption provided in sdk-scheme-adapter) refer to [deploying Mojaloop TTK simulators](https://github.com/mojaloop/helm/blob/master/mojaloop-ttk-simulators/README.md).

## 6. Known Issues

1. [#2119 - Idempotency for duplicate quote request](https://github.com/mojaloop/project/issues/2119)
2. [#2322 - Helm install failing with with "medium to large" release names](https://github.com/mojaloop/project/issues/2322)
3. [#2317 - Mojaloop Helm deployments are not compatible when deployed to ARM-arch based hosts](https://github.com/mojaloop/project/issues/2317)
4. [#2435 - Quoting-Service is incorrectly handling failed responses to FSPs when forwarding requests](https://github.com/mojaloop/project/issues/2435)
5. Test issues causing instability/intermitant failures on Test Case Results
    1. [#2717 - Thirdparty TTK Test-Collection is not repeatable](https://github.com/mojaloop/project/issues/2717)
    2. [#2925 - Helm Test Intermittent failure with 'Generic ID not found](https://github.com/mojaloop/project/issues/2925)

> ## 9. Contributors

Reviewer 2: ```suggestion
## 7. Contributors
```

# Helm Release Notes

Date | Revision | Description
---------|----------|---------
 2023-10-23 | 0 | Initial draft
 2023-10-xx | 1 | Initial release

- For *BREAKING CHANGES*, please review the section `#5` "Breaking Changes" below.
- For *KNOWN ISSUES*, please review the section `#8` "Known Issues" below.

## 0. Summary

Minor enhancements for [v15.1.0 Release](https://github.com/mojaloop/helm/blob/master/.changelog/release-v15.1.0.md), which includes the following changes:

> 1. Characterisation that improves performance and metrics around some core services: ALS, Central Ledger, Quoting Service.

Reviewer: ```suggestion
1. Performance & instrumentation improvements for the following core services: Account-Lookup-Service, Quoting-Service, ML-API-Adapter & Central Ledger.
```

> 2. NodeJS Upgrades

Reviewer: ```suggestion
2. Maintenance upgrades for all packaged services, this includes NodeJS & dependencies.
```

> 3. Bugfixes

Reviewer: ```suggestion
3. Several bug-fixes mainly due to the maintenance upgrades, and a result of the Performance Characterisation work to improve stability of Mojaloop core services under load
```

2. central-ledger: v17.0.3 -> [v17.3.0]((https://github.com/mojaloop/central-ledger/releases/tag/v17.3.0)) ([Compare](https://github.com/mojaloop/central-ledger/compare/v17.0.3...v17.3.0))
3. account-lookup-service: 14.1.0 -> [v14.2.5](https://github.com/mojaloop/account-lookup-service/releases/tag/v14.2.5) ([Compare](https://github.com/mojaloop/account-lookup-service/compare/v14.1.0...v14.2.5))
4. quoting-service: v15.0.2 -> [v15.4.0](https://github.com/mojaloop/quoting-service/releases/tag/v15.4.0) ([Compare](https://github.com/mojaloop/quoting-service/compare/v15.0.2...v15.4.0))
5. central-settlement: v15.0.0 -> [v15.0.4](https://github.com/mojaloop/central-settlement/releases/tag/v15.0.4) ([Compare](https://github.com/mojaloop/central-settlement/compare/v15.0.0...v15.0.4))
6. bulk-api-adapter: v15.0.3 -> [v16.0.1](https://github.com/mojaloop/bulk-api-adapter/releases/tag/v16.0.1) ([Compare](https://github.com/mojaloop/bulk-api-adapter/compare/v15.0.3...v16.0.1))
7. central-event-processor: [v12.0.0](https://github.com/mojaloop/central-event-processor/releases/tag/v12.0.0) *(Refer to section 5. BREAKING CHANGES)*

> 8. email-notifier: [v12.0.0](https://github.com/mojaloop/email-notifier/releases/tag/v12.0.0) *(Refer to section 5. BREAKING CHANGES)*

Reviewer: Note that we do have a new version of this service that was upgraded. However we have no way of testing it :(

9. als-oracle-pathfinder: [v12.0.0](https://github.com/mojaloop/als-oracle-pathfinder/releases/tag/v12.0.0)

> 10. transaction-requests-service: v14.0.1 -> [v14.1.1](https://github.com/mojaloop/transaction-requests-service/releases/tag/v14.1.1) ([Compare](https://github.com/mojaloop/transaction-requests-service/compare/v14.1.0...v14.1.1))

Reviewer: ```suggestion
10. transaction-requests-service: v14.1.0 -> [v14.1.1](https://github.com/mojaloop/transaction-requests-service/releases/tag/v14.1.1) ([Compare](https://github.com/mojaloop/transaction-requests-service/compare/v14.1.0...v14.1.1))
```

12. finance-portal-backend-service: [v15.0.2](https://github.com/mojaloop/finance-portal-backend-service/releases/tag/v15.0.2) *(DEPRECATED)*
13. settlement-management: [v11.0.0](https://github.com/mojaloop/settlement-management/releases/tag/v11.0.0) *(DEPRECATED)*
14. operator-settlement: [v11.0.0](https://github.com/mojaloop/operator-settlement/releases/tag/v11.0.0) *(DEPRECATED)*
15. event-sidecar: v12.0.0 -> [v13.0.1](https://github.com/mojaloop/event-sidecar/releases/tag/v13.0.1) ([Compare](https://github.com/mojaloop/event-sidecar/compare/v12.0.0...v13.0.1))
16. event-stream-processor: [v12.0.0-snapshot.7](https://github.com/mojaloop/event-stream-processor/releases/v12.0.0-snapshot.7)
17. simulator: [12.0.0](https://github.com/mojaloop/simulator/releases/tag/v12.0.0)
18. mojaloop-simulator: v14.0.1 -> [v15.0.0](https://github.com/mojaloop/mojaloop-simulator/releases/tag/v15.0.0) ([Compare](https://github.com/mojaloop/event-sidecar/compare/v14.0.1...v15.0.0))
19. sdk-scheme-adapter: v23.0.1 -> [v23.1.1](https://github.com/mojaloop/sdk-scheme-adapter/releases/tag/v23.1.1) ([Compare](https://github.com/mojaloop/sdk-scheme-adapter/compare/v23.0.1...v23.1.1))

> 20. ml-testing-toolkit: v16.1.1 -> v17.0.0(https://github.com/mojaloop/ml-testing-toolkit/releases/tag/v17.0.0) ([Compare](https://github.com/mojaloop/sml-testing-toolkit/compare/v16.1.1...v17.0.0))

Reviewer: ```suggestion
20. ml-testing-toolkit: v16.1.1 -> [v17.0.0](https://github.com/mojaloop/ml-testing-toolkit/releases/tag/v17.0.0) ([Compare](https://github.com/mojaloop/sml-testing-toolkit/compare/v16.1.1...v17.0.0))
```

> 21. ml-testing-toolkit-ui: v15.3.0 -> [v15.4.0](https://github.com/mojaloop/ml-testing-toolkit-ui/releases/tag/v15.3.0) ([Compare](https://github.com/mojaloop/ml-testing-toolkit-ui/compare/v15.3.0...v15.4.0))

Reviewer: Looks like no change here...

```suggestion
21. ml-testing-toolkit-ui: [v15.4.0](https://github.com/mojaloop/ml-testing-toolkit-ui/releases/tag/v15.3.0)
```

> 22. ml-testing-toolkit-client-lib: [v1.2.0](https://github.com/mojaloop/ml-testing-toolkit-client-lib/releases/tag/v1.2.0)

Reviewer: Was this upgrade in the Helm chart?

```suggestion
22. ml-testing-toolkit-client-lib: v1.2.0 -> [v1.2.1](https://github.com/mojaloop/ml-testing-toolkit-client-lib/releases/tag/v1.2.1)  ([Compare](https://github.com/mojaloop/ml-testing-toolkit-ui/compare/v1.2.0...v1.2.1))
```

Reviewer 3: Yes, the helm chart was updated accordingly, but image was missing the bump. Just pushed it.

23. auth-service: v14.0.0 -> [v14.0.1](https://github.com/mojaloop/auth-service/releases/tag/v14.0.1) ([Compare](https://github.com/mojaloop/auth-service/compare/v14.0.0...v14.0.1))

> 24. als-consent-oracle: v0.2.0 -> [v0.2.0](https://github.com/mojaloop/als-consent-oracle/releases/tag/v0.2.1) ([Compare](https://github.com/mojaloop/als-consent-oracle/compare/v0.2.0...v0.2.1))

Reviewer: ```suggestion
24. als-consent-oracle: v0.2.0 -> [v0.2.1](https://github.com/mojaloop/als-consent-oracle/releases/tag/v0.2.1) ([Compare](https://github.com/mojaloop/als-consent-oracle/compare/v0.2.0...v0.2.1))
```

## 8. Known Issues

1. [#2119 - Idempotency for duplicate quote request](https://github.com/mojaloop/project/issues/2119)
2. [#2322 - Helm install failing with with "medium to large" release names](https://github.com/mojaloop/project/issues/2322)
3. [#2317 - Mojaloop Helm deployments are not compatible when deployed to ARM-arch based hosts](https://github.com/mojaloop/project/issues/2317)
4. [#2892 - Disabled DFSP showing getParty info](https://github.com/mojaloop/project/issues/2892)
5. [#2435 - Quoting-Service is incorrectly handling failed responses to FSPs when forwarding requests](https://github.com/mojaloop/project/issues/2435)
6. Test issues causing instability/intermitant failures on Test Case Results
    1. [#2717 - Thirdparty TTK Test-Collection is not repeatable](https://github.com/mojaloop/project/issues/2717)
    2. [#2845 - QA: Replace Legacy-Simulator as a NORESPONSE_SIMPAYEE in Testing-Toolkit Goden Path Test-Suite](https://github.com/mojaloop/project/issues/2845)
    3. [#3027 - QA: Mojaloop Helm v14.1.0 Release - Bulk Tests fail on first run](https://github.com/mojaloop/project/issues/3027)

>     4. [#2925 - Helm Test Intermittent failure with 'Generic ID not found](https://github.com/mojaloop/project/issues/2925)

Author: @elnyry-sam-k This issue is still open.

>     4. [#2925 - Helm Test Intermittent failure with 'Generic ID not found](https://github.com/mojaloop/project/issues/2925)

Reviewer 2: yes it is.. was wondering if we saw this happen in recent times.. but lets leave it for now, thanks

1. ml-api-adapter: v14.0.0 -> [v14.0.4](https://github.com/mojaloop/ml-api-adapter/releases/tag/v14.0.4) ([Compare](https://github.com/mojaloop/ml-api-adapter/compare/v14.0.0...v14.0.4))
2. central-ledger: v17.0.3 -> [v17.3.0]((https://github.com/mojaloop/central-ledger/releases/tag/v17.3.0)) ([Compare](https://github.com/mojaloop/central-ledger/compare/v17.0.3...v17.3.0))
3. account-lookup-service: 14.1.0 -> [v14.2.5](https://github.com/mojaloop/account-lookup-service/releases/tag/v14.2.5) ([Compare](https://github.com/mojaloop/account-lookup-service/compare/v14.1.0...v14.2.5))

> 4. quoting-service: v15.0.2 -> [v15.5.0](https://github.com/mojaloop/quoting-service/releases/tag/v15.4.0) ([Compare](https://github.com/mojaloop/quoting-service/compare/v15.0.2...v15.5.0))

Reviewer 2: ```suggestion
4. quoting-service: v15.0.2 -> [v15.5.0](https://github.com/mojaloop/quoting-service/releases/tag/v15.5.0) ([Compare](https://github.com/mojaloop/quoting-service/compare/v15.0.2...v15.5.0))
```
